### PR TITLE
React-Redux Project - BookStore - Add and Remove Books

### PR DESCRIPTION
**Pull request by jmontanholi**
## In this feature branch I implemented the following:
* #### Add books:
   * #### Reducer
   * #### Action Creator
   * #### Action constant types
   * #### Buttons to dispatch actions
* #### Remove books:
   * #### Reducer
   * #### Action Creator
   * #### Action constant types
   * #### Buttons to dispatch actions

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [
+        ...state,
+        action.payload,
+      ];
+    case REMOVE_BOOK: {
+      const newArray = [...state];
+      newArray.splice((action.payload), 1);
```
Comment (vichuge) [inline src/redux/books/books.js]: - [x] Please, to delete a book use the filter method like `state.filter(book => book.id !== id)`. Thisl will help to have a better delete function in your reducer. 🐶

File: src/components/books.js [hunk 2]
```diff
@@ -1,9 +1,46 @@
...
+        <p>
+          Title:
+          { book.title }
+        </p>
+        <p>
+          Author:
+          { book.author }
+        </p>
+        <button type="button" id={books.books.indexOf(book)} onClick={(e) => { remove(e); }}>Remove Book</button>
+      </li>,
```
Comment (vichuge) [inline src/components/books.js]: - [x] Good job implementing a way to have a book, now let's restructure a little bit, why not use a `<book />` component and call it here, please inside this component add a remove function too! 🐱

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your PR messages are descriptive ✔️

## Aspects to improve ♻️

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (jmontanholi) [reply src/redux/books/books.js]: And easier too hahahah

**Review by Emmyn5600 — APPROVED** [review]
Hi, 

Good job so far! 👏🏼 @jmontanholi 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous TSE
 
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**